foundation-extensions/mobile-core/platform-services.md

```diff
@@ -1,6 +1,122 @@
+
+{% hint style="info" %}
+This feature is only available in [Android Core 1.8.0](https://aep-sdks.gitbook.io/docs/release-notes#android-core-1-8-0) or later and [iOS Core 3.0.0](https://aep-sdks.gitbook.io/docs/release-notes#adobe-experience-platform-ios-core-sdks) or later.
+{% endhint %}
+
+## Accessing Services
+
+The MobileCore extension provides a shared `ServicesProvider`, responsible for accessing the current set of provided services.
+
+### Networking
+
+The `Networking` service provides shared functionality to make asynchronous network requests and handle their responses.
+
+The following code snippet details how to make a simple network request and handle the response.
+
+{% tabs %}
+{% tab title="Android" %}
+
+```java
+AndroidNetworkService androidNetworkService = new AndroidNetworkService(ServiceProvider.getInstance().getNetworkService());
+androidNetworkService.connectUrlAsync(url,
+											  POST, payload,
+		null, 10000, 10000, new NetworkService.Callback() {
+			@Override
+			public void call(NetworkService.HttpConnection connection) {
+				// handle `httpConnection`
+			}
+});
+```
+
+{% endtab %}
+
+{% tab title="iOS(AEP 3.x)" %}
+
+```swift
+// Create your `NetworkRequest`, for more details see `NetworkRequest.swift`
+let networkRequest = NetworkRequest(url: url, httpMethod: .get, httpHeaders: headers)
+
+// Get an instance of the current network service
+let networkService = ServiceProvider.shared.networkService
+
+// Make a request
+networkService.connectAsync(networkRequest: networkRequest) { httpConnection in
+  // handle `httpConnection`
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+### DataQueuing
+
+The `DataQueuing` service provides access to FIFO queues. This service is particularly useful when used in conjunction with a `PersistentHitQueue`.
+
+The following code snippet shows how to create a `DataQueue` and add a `DataEntity` to the queue.
+
+{% tabs %}
+{% tab title="Android" %}
+
+```java
+DataQueue dataQueue = ServiceProvider.getInstance().getDataQueueService().getDataQueue(name);
+DataEntity dataEntity = new DataEntity(mydata);
+dataQueue.add(dataEntity);
+```
+
+{% endtab %}
+
+{% tab title="iOS(AEP 3.x)" %}
+
+```swift
+// Create a `DataQueue`
+guard let dataQueue = ServiceProvider.shared.dataQueueService.getDataQueue(label: name) else {
+  Log.error(label: "\(name):\(#function)", "Failed to create Data Queue")
+  return
+}
+
+// Create a `DataEntity`
+let entity = DataEntity(data: myData)
+
+// Add entity to `dataQueue`
+dataQueue.add(entity)
+```
+
+{% endtab %}
+{% endtabs %}
+
+### SystemInfoService (iOS) & DeviceInforming (Android)
+
+The `SystemInfoService` (iOS) and the `DeviceInforming`(Android) service let you access critical pieces of information related to the user's device, such as carrier name, device name, locale, and more. 
```

Comment:
I agree we don't need to have snippets for each of the systemInfoService APIs below, but we should probably at least list out each of the APIs here. And we should probably list them with their appropriate syntax. `getProperty`, `getAsset`. Debating if we should list them all out with an explanation at least but let me know your thoughts.

Reply:
@cdhoffmann address your review comments with adding a new section -> `Public Interfaces`, which follows what we did in API reference pages (https://aep-sdks.gitbook.io/docs/foundation-extensions/identity-for-edge-network/api-reference#public-classes)
Let me know if it's OK, then I can add the same thing for other Services.

Reply:
Hmmm after seeing this i think it might be too much to show the entire protocol. Especially because it can become too much if we end up doing this for all public protocols we are exposing. I wonder if we should just link to the public file in github from this section so that they can see the public APIs themselves in the source? This might be something worth discussing with the Doc team (@emdobrin @swarna04). Let me know what you think.

Reply:
The link to Swift open source repository, yes. But we can't do that for Android Platform services.